client/src/main/java/io/pravega/client/byteStream/impl/ByteStreamClientImpl.java

```diff
@@ -48,16 +48,11 @@
 
     @Override
     public ByteStreamReader createByteStreamReader(String streamName) {
-        return createByteStreamReaders(new Segment(scope, streamName, 0));
-    }
-
-    private ByteStreamReader createByteStreamReaders(Segment segment) {
-        String delegationToken = Futures.getAndHandleExceptions(controller.getOrRefreshDelegationTokenFor(segment.getScope(),
-                                                                                                          segment.getStream()
-                                                                                                                 .getStreamName()),
-                                                                RuntimeException::new);
-
-        DelegationTokenProvider tokenProvider = DelegationTokenProviderFactory.create(delegationToken, controller, segment);
+        StreamSegments segments = Futures.getThrowingException(controller.getCurrentSegments(scope, streamName));
+        Preconditions.checkState(segments.getNumberOfSegments() > 0, "Stream is sealed");
```

Comment:
Why is it an error to create a reader on a sealed stream?

Reply:
Good catch, fixed it.